test/2fa.test.js

```diff
@@ -1,257 +1,254 @@
-            .send({
-                accountId,
-                method,
-                ...(await signatureFor(accountId))
-            })
-            .expect('Content-Type', /json/)
-            .expect((res) => {
-                assert.ok(res.ok);
-                securityCode = getCodeFromLogs();
-            })
-            .expect(200);
-    });
+const VERBOSE_OUTPUT_CONFIG = {
+    ECHO_SECURITY_CODES: VERBOSE_OUTPUT || false,
+    ECHO_MESSAGE_CONTENT: VERBOSE_OUTPUT || false
+};
 
-    test('should be able to call init again with different method', async () => {
+describe('2fa method management', function () {
+    this.timeout(15000);
 
-        method = twoFactorMethods[1];
+    let app, request, testAccountHelper;
 
-        await request.post('/2fa/init')
-            .send({
-                accountId,
-                method,
-                ...(await signatureFor(accountId))
-            })
-            .expect('Content-Type', /json/)
-            .expect((res) => {
-                assert.ok(res.ok);
-                securityCode = getCodeFromLogs();
-            })
-            .expect(200);
-    });
+    before(async () => {
+        const keyStore = new nearAPI.keyStores.InMemoryKeyStore();
+        const keyPair = nearAPI.KeyPair.fromString(parseSeedPhrase(SEED_PHRASE).secretKey);
 
-    test('verify 2fa method', async () => {
+        ({ request, app } = createTestServerInstance());
+        attachEchoMessageListeners({ app, ECHO_MESSAGE_CONTENT: VERBOSE_OUTPUT_CONFIG.ECHO_MESSAGE_CONTENT });
 
-        await request.post('/2fa/verify')
-            .send({
-                accountId,
-                requestId,
-                securityCode,
-                ...(await signatureFor(accountId))
-            })
-            .expect((res) => {
-                assert.ok(res.ok);
-            })
-            .expect(200);
+        testAccountHelper = new TestAccountHelper({
+            app,
+            ECHO_SECURITY_CODES: VERBOSE_OUTPUT_CONFIG.ECHO_SECURITY_CODES,
+            keyPair,
+            keyStore,
+            request,
+        });
+
+        await models.sequelize.sync({ force: true });
     });
 
-});
+    describe('setting up 2fa method', () => {
+        let accountId;
 
-describe('after deploying contract', () => {
-    let accountId = 'testing' + Date.now();
-    let method = twoFactorMethods[0];
-    let securityCode = '';
-    let requestId = -1;
-    let testContractDeployed = true;
+        // Would prefer beforeAll, but `testContext.logs` is cleared in the global beforeEach() that would run after beforeAll here
+        beforeEach(async () => {
+            if (!accountId) {
+                accountId = await testAccountHelper.createNEARAccount();
+            }
+        });
 
-    test('initCode for an account, sets up 2fa method', async () => {
-        await createNearAccount(accountId);
+        // TODO: Should we be cleaning up accounts just in case this is running against e.g. testnet?
```

Comment:
we probably don't care much unless we want to run tests on mainnet

might be slightly easier for debugging failures if account is still around

probably best not to keep commented out code